celltype_annotation/analysis/copykat-rms-exploration.Rmd

```diff
@@ -0,0 +1,224 @@
+  system(sync_call, ignore.stdout = TRUE) 
+}
+```
+
+
+## Run CopyKAT
+
+Now, we are ready to run `copyKat` on the SCE, using the _raw expression matrix_ as described in the [GitHub `README` file](https://github.com/navinlabcode/copykat/blob/master/README.md).
+
+Of note, the `README` includes this caveat about the `"distance"` parameter:
+
+> One struggling observation is that none of one clustering method could fit all datasets. In this version, I add a distance parameters for clustering that include "euclidean" distance and correlational distance, ie. 1-"pearson" and "spearman" similarity. In general, corretional distances tend to favor noisy data, while euclidean distance tends to favor data with larger CN segments.
+
+Therefore we'll compare `"euclidian"`, `"spearman"`, and `"pearson"` values for [this parameter](https://github.com/navinlabcode/copykat/blob/b795ff793522499f814f6ae282aad1aab790902f/R/copykat.R#L251-L255).
+
+
+```{r message=FALSE, warning=FALSE}
+# Only run copykat if the final SCE does not exist
+if (file.exists(copykat_sce)) {
+  
+  sce <- readr::read_rds(copykat_sce)
+  
+} else {
+  
+  # Otherwise, read in initial SCE and run copyKat
+  sce <- readr::read_rds(local_submitter_sce_path)
+    
+  purrr::walk(
+    c("spearman", "pearson", "euclidean"), 
+    \(x) {
+      
+      copykat_result <- copykat(
+        rawmat = as.matrix(counts(sce)), 
+        id.type = "E", # we have Ensembl gene ids, not the default gene symbols ("S")
+        sam.name = params$library_id, 
+        distance = x, 
+        plot.genes = FALSE,
+        n.cores = params$threads
+      )
+      
+      # save prediction to sce
+      colname <- glue::glue("copykat_{x}")
+      colData(sce)[[colname]] <- copykat_result$prediction$copykat.pred
+      
+      # export full result
+      output_file <- file.path(output_dir,
+                              glue::glue("{params$library}_copykat_{x}.rds"))
+      readr::write_rds(copykat_result, output_file)
+    }
+  )
+  
+  # finally, export SCE with all copykat predictions
+  readr::write_rds(sce, copykat_sce)
+} 
+```
+
+## copyKat results
+
+First, we'll look at quick tables for correspondance between submitter cell type annotations and `copyKat` inferences.
+For `copyKat`, "diploid" is normal and "aneuploid" is tumor.
+
+```{r}
+table(sce$celltype, sce$copykat_spearman)
+```
+
+```{r}
+table(sce$celltype, sce$copykat_pearson)
+```
+
+```{r}
+table(sce$celltype, sce$copykat_euclidian)
+```
+
+
+Next, we'll look at some classification metrics overall, using a custom function to calculate a confusion matrix.
+For this, we will _only consider_ cells which i) the submitter was able to classify, and ii) `copyKat` was able to classify.
+In other words, all `NA` values are removed from the data before calculations.
+
+
+
+```{r}
+# Custom function to calculate some confusion matrix quantities, including
+# accuracy, TPR, and FPR
+calculate_confusion <- function(df, test_column) {
```

Comment:
Is the `test_column` supposed to be the predictions column?

Reply:
yes, I can rename!